sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -351,11 +351,12 @@ class ParquetFileFormat
       if (pushed.isDefined) {
         ParquetInputFormat.setFilterPredicate(hadoopAttemptContext.getConfiguration, pushed.get)
       }
+      val taskContext = Option(TaskContext.get())
```

Comment:
Yes, you are right. 

We have to ensure `close()` is closed at [here](https://github.com/apache/spark/pull/17436/files#diff-ee26d4c4be21e92e92a02e9f16dbc285R380) to use `OffHeapColumnVector`.